h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2301,17 +2312,26 @@ else if (gaussian.equals(_parms._family) && Link.identity.equals(_parms._link))
         }
       }
       if (_parms._compute_p_values) { // compute p-values
-        double se = 1;
+        double se = _parms._init_dispersion_factor;
         boolean seEst = false;
         double[] beta = _state.beta();
 
         if (_parms._family != binomial && _parms._family != Family.poisson) {
           seEst = true;
-          ComputeSETsk ct = new ComputeSETsk(null, _state.activeData(), _job._key, beta, _parms).doAll(_state.activeData()._adaptedFrame);
-          if (_parms._useDispersion1)
-            se = 1.0;
-          else
-            se = ct._sumsqe / (_nobs - 1 - _state.activeData().fullN());  // this is the dispersion parameter estimate
+          if (pearson.equals(_parms._dispersion_factor_mode)){
+            ComputeSETsk ct = new ComputeSETsk(null, _state.activeData(), _job._key, beta, 
+                    _parms).doAll(_state.activeData()._adaptedFrame);
+            se = ct._sumsqe / (_nobs - 1 - _state.activeData().fullN());  // this is the dispersion parameter estimate using pearson
+          } else if (ml.equals(_parms._dispersion_factor_mode)) {
+            seEst = true;
+            Log.info("estimating dispersion factor using maximum likelihood");
```

Comment:
would be good to move the info message out of the IF-statement and just do 

Log.info("estimating dispersion factor using method: " + _parms._dispersion_factor_mode);